pkg/supportbundle/collect.go

```diff
@@ -207,136 +222,394 @@ func getAnalysisFile(analyzeResults []*analyze.AnalyzeResult) (io.Reader, error)
+		Container: container,
+		Stdin:     true,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       false,
+	}, parameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(clientConfig, "POST", req.URL())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+
+	err = exec.StreamWithContext(ctx, remotecommand.StreamOptions{
+		Stdin:  bytes.NewBuffer(collectorData),
+		Stdout: stdout,
+		Stderr: stderr,
+		Tty:    false,
+	})
+
+	if err != nil {
+		return stdout.Bytes(), stderr.Bytes(), err
+	}
+	return stdout.Bytes(), stderr.Bytes(), nil
 }
 
-// collectHost runs host collectors sequentially
-func collectHost(ctx context.Context, collectSpecs []*troubleshootv1beta2.HostCollect, bundlePath string, opts SupportBundleCreateOpts, collectedData map[string][]byte) error {
-	// Run local collectors sequentially
-	for _, spec := range collectSpecs {
-		collector, ok := collect.GetHostCollector(spec, bundlePath)
-		if !ok {
-			opts.ProgressChan <- "Host collector not found"
-			continue
-		}
+func runRemoteHostCollectors(ctx context.Context, hostCollectors []*troubleshootv1beta2.HostCollect, bundlePath string, opts SupportBundleCreateOpts) (map[string][]byte, error) {
+	output := collect.NewResult()
 
-		// Send progress event: starting the collector
-		opts.ProgressChan <- fmt.Sprintf("[%s] Running host collector...", collector.Title())
+	clientset, err := kubernetes.NewForConfig(opts.KubernetesRestConfig)
+	if err != nil {
+		return nil, err
+	}
 
-		// Start a span for tracing
-		_, span := otel.Tracer(constants.LIB_TRACER_NAME).Start(ctx, collector.Title())
-		span.SetAttributes(attribute.String("type", reflect.TypeOf(collector).String()))
+	// TODO: rbac check
 
-		isExcluded, _ := collector.IsExcluded()
-		if isExcluded {
-			opts.ProgressChan <- fmt.Sprintf("[%s] Excluding host collector", collector.Title())
-			span.SetAttributes(attribute.Bool(constants.EXCLUDED, true))
-			span.End()
-			continue
-		}
+	nodeList, err := getNodeList(clientset, opts)
+	if err != nil {
+		return nil, err
+	}
+	klog.V(2).Infof("Node list to run remote host collectors: %s", nodeList.Nodes)
 
-		// Run local collector sequentially
-		result, err := collector.Collect(opts.ProgressChan)
+	// create remote pod for each node
+	labels := map[string]string{
+		"troubleshoot.sh/remote-collector": "true",
+	}
+
+	var mu sync.Mutex
+	nodeLogs := make(map[string]map[string][]byte)
+
+	ds, err := createHostCollectorDS(ctx, clientset, labels)
+	if err != nil {
+		return nil, err
+	}
+
+	// wait for at least one pod to be scheduled
+	err = waitForDS(ctx, clientset, ds)
+	if err != nil {
+		return nil, err
+	}
+
+	klog.V(2).Infof("Created Remote Host Collector Daemonset %s", ds.Name)
+	pods, err := clientset.CoreV1().Pods(ds.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector:  selectorLabelKey + "=" + selectorLabelValue,
+		TimeoutSeconds: new(int64),
+		Limit:          0,
```

Comment:
0 means that there is no limit on the number of pods returned in this list operation.  It will attempt to return all matching pods without restricting the count.